src/mac/process.rs

```diff
@@ -8,6 +8,48 @@ use std::fmt::{self, Formatter, Debug};
 use libc::{c_int, gid_t, kill, pid_t, uid_t};
 
 #[derive(Clone)]
+pub enum ProcessStatus {
+    Idle,   /* Process being created by fork. */
+    Run,    /* Currently runnable. */
+    Sleep,  /* Sleeping on an address. */
+    Stop,   /* Process debugging or suspension. */
+    Zombie, /* Awaiting collection by parent. */
+}
+
+impl ProcessStatus {
+
+    pub fn from(status:u32) -> Option<ProcessStatus> {
+        match status {
+            1 => Some( ProcessStatus::Idle ),
+            2 => Some( ProcessStatus::Run ),
+            3 => Some( ProcessStatus::Sleep ),
+            4 => Some( ProcessStatus::Stop ),
+            5 => Some( ProcessStatus::Zombie ),
+            _ => None
+        }
+    }
+
+    pub fn string<'a>(&'a self) -> &'a str {
+        match self {
+            &ProcessStatus::Idle   => "Idle",
```

Comment:
Generally we prefer to match over `*var` to avoid putting `&` before every pattern.